lighthouse-core/config/config.js

```diff
@@ -246,8 +246,24 @@ function expandArtifacts(artifacts) {
   return artifacts;
 }
 
+/**
+ * @param {LH.Flags=} flags
+ * @return {Partial<LH.Config.Settings>}
+ */
+function cleanFlagsForSettings(flags = {}) {
```

Comment:
semantically this is `removedUndefinedProperties` right?

could we call it that instead?

Reply:
not sure I'm following you, this is looking at what properties are not on `constants.defaultSettings` and dropping them from flags regardless of their value.

the JSON.parse(JSON.stringify()) approach would address the specific undefined yargs case but not an underlying problem that having miscellaneous junk present that isn't part of settings throws off our isDeepEqual later

Reply:
oh got it. 

>  what properties are not on constants.defaultSettings

ahhh.. mmhmmm. 

k lgtm